### PR TITLE
Adds Crank.Agent log file support - serilog

### PR DESCRIPTION
**Pull request by rducom**
As discussed in https://github.com/dotnet/crank/pull/365, this PR adds log file support for crank agent.
Users can pass an opt-in `--log-path` command-line argument with a full path to a log-file.

Log files roll every 1 day.

### CODE REVIEW

**Comment by rducom** [comment]
@sebastienros I have an build error related to usage of Serilog

`error SIGN001: Signing 3rd party library Serilog.dll with Microsoft certificate 'Microsoft400'. The library is considered 3rd party library due to its copyright: ''`

I wasn't expecting it, since [Serilog is used](https://github.com/microsoft/reverse-proxy/search?q=serilog) in Yarp without issues. What is Crank policy about third party lib usage ?

**Comment by sebastienros** [comment]
Needs to be registered here: https://github.com/dotnet/crank/blob/main/eng/Signing.props
And also added to https://github.com/dotnet/crank/blob/main/NOTICE.txt

File: src/Microsoft.Crank.Agent/Microsoft.Crank.Agent.csproj [hunk 1]
```diff
@@ -29,6 +29,11 @@
     <PackageReference Include="NuGet.Versioning" Version="5.5.1" />
     <PackageReference Include="Vanara.PInvoke.Kernel32" Version="3.3.14" />
     
+    <PackageReference Include="Serilog.Extensions.Logging" Version="3.0.1" />
```
Comment (sebastienros) [inline src/Microsoft.Crank.Agent/Microsoft.Crank.Agent.csproj:32]: That's a lot of direct references just for a file logger. Are they all required?

Comment (rducom) [reply src/Microsoft.Crank.Agent/Microsoft.Crank.Agent.csproj:32]: You are right, Serilog.AspNetCore includes Serilog.Extensions.Logging, so not needed

**Pull request merged**